Library/Homebrew/cmd/outdated.rb

```diff
@@ -97,6 +98,8 @@ def print_outdated(formulae_or_casks, args:)
           elsif f.head? && outdated_kegs.any? { |k| k.version.to_s == f.pkg_version.to_s }
             # There is a newer HEAD but the version number has not changed.
             "latest HEAD"
+          elsif f.tap.blank? && ENV["HOMEBREW_JSON_CORE"].present?
```

Comment:
Are you thinking just for checking outdated formulae or for the rest of it, too? If you _do_ have homebrew/core tapped and also have `HOMEBREW_JSON_CORE` set and run `brew install`, should it install from the tap over the JSON method? I think I'm leaning toward using the tap first and only defaulting back to the JSON method if the formula can't be found.

If we're prioritizing the tap for installs, I think it should be prioritized for `brew outdated`, too. It would be confusing if you ran `brew outdated` and it showed a new version, but running `brew upgrade` says the latest version is up to date.

Reply:
> Are you thinking just for checking outdated formulae or for the rest of it, too?

I'm thinking particularly for outdated formulae because you do care about getting the most up-to-date version. I appreciate it may be weird if you have `HOMEBREW_JSON_CORE` and `HOMEBREW_NO_AUTO_UPDATE` enabled, though, so unless we want to handle that special case: it can probably stay as-is.

Eventually, though, even if you have the tap tapped but `HOMEBREW_JSON_CORE` set we will probably want to stop auto-updating in favour of just pulling the JSON data instead.